src/sqlfluff/dialects/dialect_sparksql.py

```diff
@@ -1121,11 +1121,15 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
             optional=True,
         ),
         Sequence("USING", Ref("DataSourceFormatGrammar"), optional=True),
-        Ref("RowFormatClauseSegment", optional=True),
-        Ref("StoredAsGrammar", optional=True),
-        Ref("OptionsGrammar", optional=True),
-        Ref("PartitionSpecGrammar", optional=True),
-        Ref("BucketSpecGrammar", optional=True),
+        AnyNumberOf(
```

Comment:
Rather than `AnyNumberOf(..., max_times_per_element=1)` you can just use `AnySetOf`.

Reply:
In that case you could also make the `AnySetOf` grammar `optional`, and then you wouldn't need the `optional=True` on any of the inner segments.

Reply:
Got it! Thanks for the heads up on `AnySetOf` -- wasn't aware of this.